ahcore/models/jit_model.py

```diff
@@ -1,24 +1,231 @@
 from pathlib import Path
-from typing import Any
+from typing import Any, Callable
 
 import torch
 from torch.jit import ScriptModule, load
 from torch.nn import Module
 
+from ahcore.utils.types import AllModelTypes, AllOutputModes, FMEmbedType, FMOutputMode, JitModelType, JitOutputMode
+
 
 class AhcoreJitModel(ScriptModule):
-    def __init__(self, model: ScriptModule) -> None:
+    """
+    Base class for the jit compiled models in Ahcore.
+    """
+    def __init__(self, model: ScriptModule, model_type: AllModelTypes, output_mode: AllOutputModes) -> None:
+        """
+        Constructor for the AhcoreJitModel class.
+
+        Parameters
+        ----------
+        model: ScriptModule
+            The jit compiled model.
+
+        model_type: AllModelTypes
+            The type of the model.
+
+        output_mode: AllOutputModes
+            The output mode of the model. This is used to determine the forward function of the model.
+
+        Returns
+        -------
+        None
+        """
         super().__init__()  # type: ignore
         self._model = model
+        self._model_type = model_type
+        self._output_mode = output_mode
+        self._forward_function: Callable[[torch.Tensor], torch.Tensor]
```

Comment:
Bit confused about this class now, is it perhaps supposed to be an abstract class? It seems the from jit path and forwards are also implemented in the child classes. Either this one should have abstract methods or the children shouldn't have to implement it.

Reply:
Thank you. I will implement this